codec/src/codec.rs

```diff
@@ -147,6 +147,17 @@ pub trait EncodeFixed: Write + FixedSize {
 // Automatically implement `EncodeFixed` for types that implement `Write` and `FixedSize`.
 impl<T: Write + FixedSize> EncodeFixed for T {}
 
+/// Convenience trait combining [EncodeFixed] and [Decode].
+///
+/// Represents types that can be both fully encoded and decoded from a fixed-size byte sequence.
+pub trait CodecFixed:
```

Comment:
With all these extra trait requirements, this probably isn't the right place for this but this is what is required to remove `Array ` from value storage.